### PR TITLE
(GH-1702) Improve documentation for empty switches

### LINKED ISSUE #1702: Improve documentation for passing empty values for an argument
To pass empty values for arguments to Cake 1.0 a space needs to be used instead of `=`. We already mention it in [upgrade instructions](https://cakebuild.net/docs/getting-started/upgrade#passing-empty-arguments), but could make it more clear that it also applies for empty values (`--foo=""`).

Additionally we should document this on the runner page.

See also https://github.com/cake-build/cake/discussions/3198

### PR DESCRIPTION
**Pull request by pascalberger**
Improve documentation for switches without or with and empty value.

Fixes #1702

### CODE REVIEW

File: input/Shared/switches.txt [hunk 1]
```diff
@@ -32,3 +32,11 @@
 
 All switches not recognized by Cake will be added to an argument list that is passed to the build script.
 See [Arguments And Environment Variables](/docs/writing-builds/args-and-environment-vars#arguments) how to read arguments in your script.
+
+:::{.alert .alert-info}
+Use a space instead of `=` if the value of a switch can be empty or null:
```
Comment (nils-a) [inline input/Shared/switches.txt:37]: Is "empty or null" precice enough? My tests always result in `null`, regardless whether I set `--foo` or `--foo ""`.

**Review by augustoproiete — APPROVED** [review]
LGTM

File: input/Shared/switches.txt (hunk 1 above)
Comment (augustoproiete) [reply input/Shared/switches.txt:37]: I think it's clear, and the code example after that helps. How would phrase it differently?

Also, what do you mean by "_my tests always result in `null`_"

In my tests, both `--foo` and `--foo ""` results in the argument not being present in the collection of arguments, which means that `Argument<string>("foo")` throws an exception on both cases with `Argument 'foo' was not set`.

So the only way to get `null` is to specify `null` as the default value i.e. `Argument<string>("foo", null)` but in that case you're only getting `null` because you said that was the default... Not because Cake decided for you. The same way that `Argument<string>("foo", "abc")` will give you `abc` for both `--foo` and `--foo ""`

**Review by nils-a — APPROVED** [review]
LGTM
